example/symmetric_difference.cpp

```diff
@@ -5,10 +5,14 @@
 #include <boost/hana/assert.hpp>
```

Comment:
I thought of that, too. I chose this way because we don't have different implementations for set and map.